lib/cylc/cycling/iso8601.py

```diff
@@ -288,9 +293,12 @@ class ISO8601Exclusions(ExclusionBase):
     grouped exclusion sequences. The Python ``in`` and ``not in`` operators
     may be used on this object to determine if a point is in the collection
     of exclusion sequences."""
+
+    __slots__ = ExclusionBase.__slots__ + ('p_iso_exclusions',)
+
     def __init__(self, excl_points, start_point, end_point=None):
         super(ISO8601Exclusions, self).__init__(start_point, end_point)
-        self.p_iso_exclusions = set()
+        self.p_iso_exclusions = []
```

Comment:
What's the rationale behind this change?  I would have thought `set` was better, with the need for membership testing below...